pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -84,16 +88,36 @@ func validateIngressNetworking(networking *mesh_proto.Dataplane_Networking) vali
 	return err
 }
 
-func validateInbound(inbound *mesh_proto.Dataplane_Networking_Inbound, dpAddress string) validators.ValidationError {
+func validateInbound(inbound *mesh_proto.Dataplane_Networking_Inbound, dpAddress string, isTransparenProxy bool) validators.ValidationError {
 	var result validators.ValidationError
 	if inbound.Port < 1 || inbound.Port > 65535 {
 		result.AddViolationAt(validators.RootedAt("port"), `port has to be in range of [1, 65535]`)
 	}
 	if inbound.ServicePort > 65535 {
 		result.AddViolationAt(validators.RootedAt("servicePort"), `servicePort has to be in range of [0, 65535]`)
 	}
-	if inbound.Address != "" && net.ParseIP(inbound.Address) == nil {
-		result.AddViolationAt(validators.RootedAt("address"), `address has to be valid IP address`)
+	if inbound.ServiceAddress != "" {
+		if isTransparenProxy {
+			result.AddViolationAt(validators.RootedAt("serviceAddress"), `serviceAddress can not be set in Transparent Proxy mode`)
+		}
```

Comment:
I see no point in validating that there is transparent proxy in place. Technically it's not even a problem (stated by Ilya) because the request will be sent by Envoy container which is excluded from iptables.